### PR TITLE
Add type inference support for tuples.

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**
Fixes https://github.com/dotnet/roslyn/issues/18969

### CODE REVIEW

**Comment by CyrusNajmabadi** [comment]
Tagging @dotnet/roslyn-ide

File: src/EditorFeatures/CSharpTest/Diagnostics/GenerateMethod/GenerateMethodTests.cs [hunk 1]
```diff
@@ -7379,5 +7379,67 @@ public void Foo()
     }
 }");
         }
+
```
Comment (rchande) [inline src/EditorFeatures/CSharpTest/Diagnostics/GenerateMethod/GenerateMethodTests.cs]: No tests at the type inferrer level?

**Comment by CyrusNajmabadi** [comment]
Tagging @MattGertz 


**Customer scenario**

user generates code in the context of a tuple. That code gets typed as 'object' instead of the appropriate element type.  user is unhappy :(

**Bugs this fixes:**

#18969

**Workarounds, if any**

None.

**Risk**

Low.

**Performance impact**

Low.

**Is this a regression from a previous update?**

No.  Tuples are new.  We just didn't put in all the necessary IDE support to handle them well.

**How was the bug found?**

Found while using the product.

**Comment by CyrusNajmabadi** [comment]
retest windows_release_vs-integration_prtest please

**Comment by jcouv** [comment]
Thanks! 👍

**Pull request merged**